homeassistant/components/livisi/shc.py

```diff
@@ -1,106 +1,52 @@
 """Code for the Livisi Smart Home Controller."""
-import asyncio
-from collections.abc import Callable
-import json
 from typing import Any
 
-from aiolivisi import AioLivisi, Websocket
+from aiolivisi import AioLivisi, LivisiEvent, Websocket
 
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.core import HomeAssistant
-from homeassistant.helpers import aiohttp_client
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import dispatcher_send
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
 
 from .const import (
     AVATAR_PORT,
     CLASSIC_PORT,
-    DEVICE_POLLING_DELAY,
+    LIVISI_DISCOVERY_NEW,
+    LIVISI_REACHABILITY_CHANGE,
+    LIVISI_STATE_CHANGE,
     PSS_DEVICE_TYPE,
-    SWITCH_PLATFORM,
 )
+from .coordinator import LivisiDataUpdateCoordinator
 
 
-class SHC:
+class SHC(CoordinatorEntity[LivisiDataUpdateCoordinator]):
```

Comment:
I'd merge this class into the coordinator class.

Reply:
The SHC is not a coordinator entity. Please read more about our architecture. I don't think you understand it.

https://developers.home-assistant.io/docs/architecture_components/

https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities